tasks/sysctl.yml

```diff
@@ -17,9 +17,20 @@
     value: "{{ sysctl_net_ipv6_conf_accept_ra_rtr_pref|int }}"
     state: present
     sysctl_set: 'yes'
+  when: system_has_ipv6
   tags:
     - sysctl
 
+- name: Sysctl rules if the system has ipv6
+  ansible.builtin.set_fact:
+    sysctl_settings: "{{ generic_sysctl_settings | combine(ipv4_sysctl_settings) | combine(ipv6_sysctl_settings) }}"
+  when: system_has_ipv6
+
+- name: Sysctl rules if the system does not have ipv6
+  ansible.builtin.set_fact:
+    sysctl_settings: "{{ generic_sysctl_settings | combine(ipv4_sysctl_settings) }}"
+  when: not system_has_ipv6
+
 - name: configure sysctl
```

Comment:
and this will be a unused since `generic_sysctl_settings`,  `ipv4_sysctl_settings`  and `ipv6_sysctl_settings`  replaces it.

Reply:
It is indeed unused outside this playbook, which was previously also the case.

The final variable has to be set up before feeding the settings to `sysctl` in the part below this, which is why this is here. I would argue that it makes the intent very readable, merging it into a variable, with a boolean condition, and then using that variable later in the playbook.

Reply:
I meant line 34:85
```yaml
  ansible.posix.sysctl:
    name: "{{ item.key }}"
    value: "{{ item.value|int }}"
    state: present
    sysctl_set: 'yes'
  with_dict: "{{ sysctl_settings }}"
```

Reply:
nvm me, the `sysctl_settings` are a fact and get looped in line 34.